assert/assertions.go

```diff
@@ -522,29 +528,40 @@ func NotSame(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}
 		h.Helper()
 	}
 
-	if samePointers(expected, actual) {
-		return Fail(t, fmt.Sprintf(
-			"Expected and actual point to the same object: %p %#v",
-			expected, expected), msgAndArgs...)
+	same, ok := samePointers(expected, actual)
+
+	if ok {
```

Comment:
Almost there, if you flip this to `if !ok {` then you can avoid the nested if statements and fully preserve line of sight.

Reply:
Also we don't really want the blank line first as this is "error handling" for the line above.

Reply:
I've made the changes